### ISSUE TITLE
Can't restart SM after log rolls

### ISSUE THREAD

**Issue by kur3us** [Netflix/security_monkey#805]
The supervisor can't restart SecurityMonkey after the log rolls.
Error in supervisor log:

`ValueError: Unable to configure handler 'file': [Errno 13] Permission denied: '/var/log/security_monkey/securitymonkey.log'`

When the log rolls to .1, the new log file is created with ubuntu:ubuntu permission instead of ubuntu:www-data

```
ubuntu@ip-10-0-1-56:/var/log/security_monkey$ ls -alrt
total 10544
drwxrwxr-x 12 root   syslog       4096 Aug 25 01:17 ..
-rw-rw-r--  1 ubuntu www-data 10485725 Aug 25 17:22 securitymonkey.log.1
-rw-rw-r--  1 ubuntu ubuntu        159 Aug 25 17:23 securitymonkey.log
drwxr-xr-x  2 ubuntu www-data     4096 Aug 25 17:23 .
-rw-r--r--  1 root   root         2273 Aug 25 20:37 security_monkey.error.log
-rw-r--r--  1 root   root       284133 Aug 25 20:37 security_monkey.access.log
```

I believe this is causing the error. If I rerun the command from the setup instructions:

`sudo chown -R `whoami`:www-data /var/log/security_monkey/`

then it can restart without a problem.

I believe a `sudo chmod g+s /var/log/security_monkey` will help.

**Comment by monkeysecurity** [comment]
I believe @mcpeak did some work in this area.  Tagging him for any comments. I think there was a modification to the flask logger that wold set group permissions or something.

**Comment by mcpeak** [comment]
In config you should be able to use `logging.handlers.GroupWriteRotatingFileHandler` instead of `logging.handlers.RotatingFileHandler`.  This will make it so the newly created files have group-writable, which should mitigate this problem.  Let me know if this doesn't work.

**Comment by monkeysecurity** [comment]
@kur3us - Are you using the quickstart setup or did you customize the permissions to have it run in the www-data group?

**Comment by kur3us** [comment]
I used the quickstart setup.

**Comment by kur3us** [comment]
I'm also seeing errors like this in /var/log/supervisor/securitymonkeyscheduler-stderr---supervisor-GE6Imt.log:

```
Logged from file watcher.py, line 626
Traceback (most recent call last):
  File "/usr/lib/python2.7/logging/handlers.py", line 77, in emit
    self.doRollover()
  File "/usr/local/src/security_monkey/security_monkey/__init__.py", line 254, in doRollover
    handlers.RotatingFileHandler.doRollover(self)
  File "/usr/lib/python2.7/logging/handlers.py", line 136, in doRollover
    os.rename(sfn, dfn)
OSError: [Errno 13] Permission denied
Logged from file peering.py, line 68
Traceback (most recent call last):
  File "/usr/lib/python2.7/logging/handlers.py", line 77, in emit
    self.doRollover()
  File "/usr/local/src/security_monkey/security_monkey/__init__.py", line 254, in doRollover
    handlers.RotatingFileHandler.doRollover(self)
  File "/usr/lib/python2.7/logging/handlers.py", line 136, in doRollover
    os.rename(sfn, dfn)
OSError: [Errno 13] Permission denied
Logged from file peering.py, line 62
```

**Comment by kur3us** [comment]
With respect to  @mcpeak 's comment, it looks like the config file IS using logging.handlers.GroupWriteRotatingFileHandler.

**Comment by mcpeak** [comment]
OK, since group write is on, you should be able to make this work if you add `ubuntu` as a member of `www-data`.  The GroupWriteRotatingFileHandler makes it so that you have g+w.

**Comment by kur3us** [comment]
@mcpeak, I added ubuntu to the group. There is one more thing that needs to be done. In the quickstart guide, it says to run these commands:

```
sudo mkdir /var/log/security_monkey
sudo chown -R `whoami`:www-data /var/log/security_monkey/
```

That creates a directory with these permissions:

`drwxr-xr-x  2 ubuntu www-data 4096 Aug 31 03:53 .`

The log rolling moves the current file to a .1, and then creates a new file that looks like this:

`-rw-rw-r-- 1 www-data www-data  1273655 Aug 31 04:01 securitymonkey.log`

So the Security Monkey process is running as "www-data", but the directory is owned by "ubuntu", and the group permission for write is not set.

So I think this command must also be done:

`chmod 775 /var/log/security_monkey/
`
As soon as I did that, the errors in the supervisor log stopped, and a the securitymonkey logs rolled successfully.
